### PR TITLE
Add simple load test CLI for remixing a project

### PR DESCRIPTION
**Pull request by humphd**
This simulates what happens when Thimble remixes a project, loading the files (tarball), project metadata, and a default HTML file.  You can run it like this:

```
node ./scripts/load-test-remix.js --url https://thimble.mozilla.org -i 1235
Running load test for host=https://thimble.mozilla.org and id=1235
Step 1: Loading project tarball
  Started
  Completed. Loaded 34.3 kB in 756ms.
Step 2: Loading project metadata
  Started
  Completed. Loaded 940 B in 620ms.
Step 3: Loading default HTML file
  Started
  Completed. Loaded 359 B in 224ms.
Finished all steps in 1.6s.
```

My thought is that we try and run a bunch of these at once somehow, and push the server.  I'm not sure what a good default project ID would be for each of staging and prod, esp. one that has a lot of files.

### CODE REVIEW

**Comment by humphd** [comment]
Here is what that [project we were debugging](https://thimbleprojects.org/lossendiere/253024/) yesterday does:

```
node ./scripts/load-test-remix.js --url https://thimble.mozilla.org -i 253024
Running load test for host=https://thimble.mozilla.org and id=253024
Step 1: Loading project tarball
  Started
  Completed. Loaded 1.86 MB in 1.9s.
Step 2: Loading project metadata
  Started
  Completed. Loaded 3.67 kB in 999ms.
Step 3: Loading default HTML file
  Started
  Completed. Loaded 359 B in 114ms.
Finished all steps in 3s.
```

**Comment by humphd** [comment]
Running a simple stress test (e.g., starting many instances of this) is easy with `pm2`:

```
npm install pm2
pm2 start ./scripts/load-test-remix.js -i 4 -- --url https://thimble.mozilla.org -i 253024
```

Here I'm asking it to start 4 instances, but you could do any number.  It will run the script, then when it exists/errors, start it again, and run like that forever.

Doing this for a few seconds on my dev machine against prod, did this:

screen shot 2017-05-30 at 3 54 38 pm

File: package.json [hunk 1]
```diff
@@ -86,5 +86,11 @@
     "invalidate": "node scripts/invalidate.js",
     "localize": "node scripts/properties2json.js",
     "localize-client": "node scripts/localize-client.js"
+  },
+  "devDependencies": {
+    "commander": "^2.9.0",
+    "pretty-bytes": "^4.0.2",
+    "pretty-ms": "^2.1.0",
+    "request": "^2.80.0"
```
Comment (gideonthomas) [inline package.json]: this is already a dependency

File: scripts/load-test-remix.js [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+ * default to staging.  You must specify an --id (or -i) for the remix ID.
+ */
+
+"use strict";
+
+let program = require("commander");
+let prettyBytes = require("pretty-bytes");
+let prettyMs = require("pretty-ms");
+let request = require("request").defaults({
+    headers: {
```
Comment (gideonthomas) [inline scripts/load-test-remix.js]: two space indents in Thimble :D

File: scripts/load-test-remix.js [hunk 3]
```diff
@@ -0,0 +1,114 @@
+/**
+ * Simulate what Thimble does when a project is remixed. This is a simplification
```
Comment (gideonthomas) [inline scripts/load-test-remix.js]: can we add this to the `program.` chain below in a function call to `description()`?

Kinda like https://github.com/mozilla/publish.webmaker.org/blob/master/scripts/export.js#L98

File: scripts/load-test-remix.js [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+});
+
+program
+    .option("-u, --url [url]", "Host URL", "https://bramble.mofostaging.net")
+    .option("-i, --id <n>", "Remix Project ID")
+    .parse(process.argv);
+
+if(!program.url || !program.id) {
+    program.help();
+    process.exit(1);
```
Comment (gideonthomas) [inline scripts/load-test-remix.js]: I don't think you need to do this, I think it exits immediately inside `help()`

File: scripts/load-test-remix.js [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+    return;
+}
+
+let host = program.url.replace(/\/$/, "");
+let id = program.id
```
Comment (gideonthomas) [inline scripts/load-test-remix.js]: semicolon

File: scripts/load-test-remix.js [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+let totalTimeMS = 0;
+
+console.log("Running load test for host=%s and id=%s", host, id);
+
+/**
+ * 1) Load project tarball
+ * GET arraybuffer {host}/projects/{remixID or ID}/files/data?cacheBust={Date.now()}
```
Comment (gideonthomas) [inline scripts/load-test-remix.js]: I'm pretty sure this is wrong. The param here is ONLY `id`. It cannot take `remixID` unless the url is suffixed with `/remix`

File: scripts/load-test-remix.js [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+ */
+function step1() {
+    console.log("Step 1: Loading project tarball");
+    console.log("  Started");
+
+    request.get({
+        encoding: null,
+        url: host + "/projects/" + id + "/files/data?cacheBust=" + Date.now()
```
Comment (gideonthomas) [inline scripts/load-test-remix.js]: How is this working without auth?

**Comment by humphd** [comment]
Pushed a new commit with all the changes you suggested.  I don't think any of this *should* need auth, since I followed the flow for an anonymous user.  We can add more tests like this for an authenticated user, which would stress the publish stuff even more.  But this gets us started with the least fuss.

**Pull request closed without merging**